packages/fast-components-react-msft/src/carousel/carousel.tsx

```diff
@@ -0,0 +1,227 @@
+import * as React from "react";
+import Foundation, { HandledProps } from "@microsoft/fast-components-foundation-react";
+import {
+    CarouselHandledProps,
+    CarouselProps,
+    CarouselUnhandledProps,
+    Slide,
+    SlideTheme,
+} from "./carousel.props";
+import { Flipper, FlipperDirection } from "../flipper";
+import { Tabs } from "@microsoft/fast-components-react-base";
+import { TabsClassNameContract } from "@microsoft/fast-components-class-name-contracts-base";
+import { get } from "lodash-es";
+
+/**
+ * The carousel state interface
+ */
+export interface CarouselState {
+    /**
+     * Holds the active tab id to share with other controls
+     */
+    activeId: string;
+}
+
+class Carousel extends Foundation<
+    CarouselHandledProps,
+    CarouselUnhandledProps,
+    CarouselState
+> {
+    public static displayName: string = "Carousel";
+
+    protected handledProps: HandledProps<CarouselHandledProps> = {
+        managedClasses: void 0,
+        label: void 0,
+        activeId: void 0,
+        items: void 0,
+    };
+
+    private slides: Slide[];
```

Comment:
Can you elaborate on the decision to store the slides on the `this` object of the class?

Reply:
Having a slides object allows several things.
- It enforces an array of type Slide
- It means we don't have to check for an array or type each time we manipulate the slides

Reply:
Left a few comments related to this below.